torch/fx/experimental/migrate_gradual_types/constraint_generator.py

```diff
@@ -0,0 +1,334 @@
+import torch
+import operator
+from typing import Callable, Dict
+from torch.fx.experimental.migrate_gradual_types.constraint import ApplyBroadcasting, CalcProduct  # type: ignore[import]
+from torch.fx.experimental.migrate_gradual_types.constraint import Disj, TGreatestUpperBound  # type: ignore[import]
+from torch.fx.experimental.migrate_gradual_types.constraint import CalcMaxPool, CalcConv, Conj  # type: ignore[import]
+from torch.fx.experimental.migrate_gradual_types.constraint import BinConstraintT, CanReshape  # type: ignore[import]
+from torch.fx.experimental.migrate_gradual_types.constraint import BinConstraintD  # type: ignore[import]
+from torch.fx.experimental.migrate_gradual_types.operation import op_eq, op_matching  # type: ignore[import]
+from torch.fx.experimental.migrate_gradual_types.operation import op_consistency, op_leq  # type: ignore[import]
+from torch.fx.experimental.migrate_gradual_types.operation import op_precision  # type: ignore[import]
+from torch.fx.node import Target, Node
+from torch.fx.experimental.migrate_gradual_types.util import gen_tensor_dims, gen_nat_constraints  # type: ignore[import]
+from torch.fx.experimental.migrate_gradual_types.util import gen_dvar, gen_tvar  # type: ignore[import]
+from torch.fx.tensor_type import Dyn, TensorType
+from torch.nn.modules.conv import Conv2d
+from torch.nn.modules.batchnorm import BatchNorm2d
+_INFERENCE_RULES: Dict[Target, Callable] = {}
```

Comment:
nit: add some vertical whitespace before this

Reply:
added